samples/cpp/tvl1_optical_flow.cpp

```diff
@@ -148,23 +148,33 @@ static void writeOpticalFlowToFile(const Mat_<Point2f>& flow, const string& file
 
 int main(int argc, const char* argv[])
 {
-    if (argc < 3)
+    cv::CommandLineParser parser(argc, argv, "{help h || show help message}"
+            "{ frame0 | | frame 0}{ frame1 | | frame 1}{ @input | | output flow}");
+    if (parser.has("help"))
     {
-        cerr << "Usage : " << argv[0] << "<frame0> <frame1> [<output_flow>]" << endl;
+        parser.printMessage();
+        return 0;
+    }
+    string frame0_name = parser.get<string>("frame0");
+    string frame1_name = parser.get<string>("frame1");
+    string file = parser.get<string>("@input");
```

Comment:
It is better to rename `@input` to `@output` and keep using positional arguments for `frame0` and `frame1` and check input frames names for emptiness.